src/native/corehost/bundle/extractor.cpp

```diff
@@ -103,9 +103,11 @@ void extractor_t::extract(const file_entry_t &entry, reader_t &reader)
 {
     FILE* file = create_extraction_file(entry.relative_path());
     reader.set_offset(entry.offset());
-    size_t size = entry.size();
+    int64_t size = entry.size();
+    assert(size < static_cast<int64_t>(std::numeric_limits<size_t>::max()));
```

Comment:
Nit: If this is going to be a pattern (see also changes in _reader.h_, _corehost\_init.cpp_), does it make sense to extract this out into a helper function or macro?

Something like (pseudo-code):

```c
size_t toSizeT_dbgchecked<T>(T input)
{
    assert(input >= 0);
    assert(std::type_traits::make_unsigned<T>(input) <= size_t::max);
    return static_cast<size_t>(input);
}
```

Reply:
Sure.